File: src/main/java/com/hedvig/backoffice/services/account/dto/AccountEntryRequestDTO.java
package com.hedvig.backoffice.services.account.dto;

import com.hedvig.backoffice.graphql.types.account.AccountEntryInput;
import lombok.Value;

import javax.money.MonetaryAmount;
import java.time.LocalDate;
import java.util.Optional;
import java.util.UUID;

@Value
public class AccountEntryRequestDTO {
  LocalDate fromDate;
  MonetaryAmount amount;
  AccountEntryType type;
  String source;
  String reference;
  Optional<UUID> matchesEntryId;
  Optional<String> title;
  Optional<String> comment;
  String addedBy;

  public static AccountEntryRequestDTO from(final AccountEntryInput accountEntryInput, final String addedBy) {
    return new AccountEntryRequestDTO(
      accountEntryInput.getFromDate(),
      accountEntryInput.getAmount(),
      accountEntryInput.getType(),
      accountEntryInput.getSource(),
      accountEntryInput.getReference(),
      Optional.empty(),
      accountEntryInput.getTitle(),
      accountEntryInput.getComment(),
      addedBy
    );
  }
}

File: src/main/java/com/hedvig/backoffice/services/account/dto/AccountEntryRequestDTO.kt
package com.hedvig.backoffice.services.account.dto

import com.hedvig.backoffice.graphql.types.account.AccountEntryInput
import java.time.LocalDate
import java.util.UUID
import javax.money.MonetaryAmount

data class AccountEntryRequestDTO(
    val fromDate: LocalDate,
    val amount: MonetaryAmount,
    val type: String,
    val source: String,
    val reference: String,
    val matchesEntryId: UUID?,
    val title: String?,
    val comment: String?,
    val addedBy: String
) {
    companion object {
        fun from(accountEntryInput: AccountEntryInput, addedBy: String): AccountEntryRequestDTO {
            return AccountEntryRequestDTO(
                fromDate = accountEntryInput.fromDate,
                amount = accountEntryInput.amount,
                type = accountEntryInput.type,
                source = accountEntryInput.source,
                reference = accountEntryInput.reference,
                matchesEntryId = null,
                title = accountEntryInput.title,
                comment = accountEntryInput.comment,
                addedBy = addedBy
            )
        }
    }
}
